### PR TITLE
fix(datepicker): use 3 rows to display months of year (consistent with internal mocks)

### LINKED ISSUE #5202: Datepicker month names are over each other in year/month selection
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
The date picker should be pleasantly displayed, with some padding between month names. The width of the picker should be what ever the minimum is to display the data in the given locale.

#### What is the current behavior?
nayttokuva 2017-06-19 kello 10 02 44

#### What are the steps to reproduce?
View the date picker https://material.angular.io/components/component/datepicker in finnish locale (FI-fi), activate the date picker dialog, click the month / year title. 

nayttokuva 2017-06-19 kello 10 10 45

#### What is the use-case or motivation for changing an existing behavior?
User experience is not what it should be.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular 4.1.3, Material 2.0.0-beta.6, Mac Sierra, TS 2.3.3, all browsers, screen shot from Chrome 58.0.3029.110 (64-bit)

### PR DESCRIPTION
**Pull request by mmalerba**
fixes #5202

### CODE REVIEW

File: src/lib/datepicker/calendar-body.scss [hunk 1]
```diff
@@ -1,12 +1,12 @@
 $mat-calendar-body-label-padding-start: 5% !default;
-$mat-calendar-body-label-translation: -6px !default;
+// We don't want the label to jump around when we switch between month and year views, so we use
+// the same amount of padding regardless of the number of columns.
+$mat-calendar-body-label-side-padding: 33% / 7 !default;
```
Comment (jelbourn) [inline src/lib/datepicker/calendar-body.scss]: Can you explain where the `33%` comes from?

File: src/lib/datepicker/calendar.scss [hunk 2]
```diff
@@ -32,8 +32,8 @@ $mat-calendar-next-icon-transform: translateX(-2px) rotate(45deg);
 
 .mat-calendar-controls {
   display: flex;
-  padding: $mat-calendar-controls-vertical-padding $mat-calendar-controls-end-padding
-           $mat-calendar-controls-vertical-padding $mat-calendar-controls-start-padding;
+  margin: $mat-calendar-controls-vertical-padding $mat-calendar-controls-side-margin
+           $mat-calendar-controls-vertical-padding $mat-calendar-controls-side-margin;
```
Comment (jelbourn) [inline src/lib/datepicker/calendar.scss]: I think you can simplify this to just 
```scss
margin: $mat-calendar-controls-vertical-padding $mat-calendar-controls-side-margin;
```

File: src/lib/datepicker/calendar-body.html [hunk 3]
```diff
@@ -3,7 +3,12 @@
   aria-hidden because we don't want it to be read out as one of the weeks in the month.
 -->
 <tr *ngIf="_firstRowOffset < labelMinRequiredCells" aria-hidden="true">
-  <td class="mat-calendar-body-label" [attr.colspan]="numCols" >{{label}}</td>
+  <td class="mat-calendar-body-label"
```
Comment (jelbourn) [inline src/lib/datepicker/calendar-body.html]: Add comments explaining the padding calculations?

File: src/lib/datepicker/calendar-body.ts [hunk 4]
```diff
@@ -68,6 +68,8 @@ export class MdCalendarBody {
   /** The cell number of the active cell in the table. */
   @Input() activeCell = 0;
 
+  @Input() cellAspectRatio = 1;
```
Comment (jelbourn) [inline src/lib/datepicker/calendar-body.ts]: Add comment explaining how and why this is used?

**Comment by mmalerba** [comment]
comments addressed, PTAL

**Pull request merged**